lib/seahorn/HornCex.cc

```diff
@@ -220,11 +227,25 @@ namespace seahorn
               << ". Expected .xml, .ll, or .bc.\n";
     }
 
-    if (!SliceOutputFile.empty()) {
-      sliceErrorTrace(F, trace);
-      dumpLLVMBitcode(M, SliceOutputFile.c_str());
+    if (!BmcSliceOutputFile.empty()) {
+      llvm::DenseSet<const llvm::BasicBlock*> region(trace.getRegion());
+      reduceToRegion(F, region, false);
+      dumpLLVMBitcode(M, BmcSliceOutputFile.c_str());
       return true;
     }
+
+    if (!CpSliceOutputFile.empty())
+    {
+      DenseSet<const BasicBlock*> region;
+      for (const CutPoint* cp:cpTrace)
+      for (CutPoint::const_iterator it=cp->succ_begin();it!=cp->succ_end();it++)
+      for (CpEdge::iterator ite=(*it)->begin();ite!=(*it)->end();ite++)
+        region.insert(&*ite);
+      reduceToRegion(F,region,false);
```

Comment:
I'd like to get to the bottom of it. If the function fails here, it might be failing somewhere else as well, but we have not noticed.

Can you send me a failing test case so that I can debug it as well?